ruby/ql/lib/codeql/ruby/frameworks/Twirp.qll

```diff
@@ -0,0 +1,86 @@
+/**
+ * Provides classes for modeling the `Twirp` framework.
+ */
+
+private import codeql.ruby.DataFlow
+private import codeql.ruby.CFG
+private import codeql.ruby.ApiGraphs
+private import codeql.ruby.AST as Ast
+private import codeql.ruby.security.ServerSideRequestForgeryCustomizations
+private import codeql.ruby.Concepts
+
+/**
+ * Provides classes for modeling the `Twirp` framework.
+ */
+module Twirp {
+  /**
+   * A Twirp service instantiation
+   */
+  class ServiceInstantiation extends DataFlow::CallNode {
```

Comment:
A lot of the predicates in this class feel like implementation details - I'd suggest making all of them `private` except for `getHandlerMethod()` as we're committing to maintaining any public methods as part of the public API.

Reply:
Will do, thanks!